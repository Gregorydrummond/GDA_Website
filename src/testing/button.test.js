import React from 'react'
import ReactDOM from 'react-dom'
import Button from 'react-bootstrap/Button'
import { render } from '@testing-library/react'
import { create } from 'react-test-renderer'
import '@testing-library/jest-dom/extend-expect'

describe("Buttons", () => {

    test("Buttons Render Without Crashing", () => {
        const div = document.createElement("div");
        ReactDOM.render(<Button></Button>, div);
    })
    
    test("Renders Play Button Correctly", () => {
        const button = create(<Button>Play</Button>);
        expect(button.toJSON()).toMatchSnapshot();
    })

})

