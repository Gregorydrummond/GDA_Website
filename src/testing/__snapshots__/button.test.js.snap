// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons Renders Play Button Correctly 1`] = `
<button
  className="btn btn-primary"
  disabled={false}
  type="button"
>
  Play
</button>
`;
